File: ToDo/ToDo.Core/Entities/ToDoItem.cs

namespace ToDo.Core
{
    public class ToDoItem : BaseEntity
    {
        public string Title { get; set; } = null!;
        public string? Description { get; set; }
        public DateTime DueDate { get; set; }
        public bool IsDone { get; set; }

        public int UserId { get; set; }
        public User User { get; set; } = null!;

    }
}

File: ToDo/UI/todo_application/lib/models/to_do_item.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:todo_application/models/user.dart';

part 'to_do_item.g.dart';

@JsonSerializable()
class ToDoItem {
  int? id;
  String? title;
  String? description;
  DateTime? dueDate;
  bool? isDone;
  int? userId;
  User? user;

  ToDoItem(this.id, this.title, this.description, this.dueDate, this.isDone, this.userId, this.user);

  factory ToDoItem.fromJson(Map<String, dynamic> json) =>
      _$ToDoItemFromJson(json);

  Map<String, dynamic> toJson() => _$ToDoItemToJson(this);
}
